framework/test/gtest/BundleTrackerMethodTest.cpp

```diff
@@ -0,0 +1,480 @@
+#include "cppmicroservices/Bundle.h"
+#include "cppmicroservices/BundleTracker.h"
+#include "cppmicroservices/Framework.h"
+#include "cppmicroservices/FrameworkEvent.h"
+#include "cppmicroservices/FrameworkFactory.h"
+
+#include "TestUtils.h"
+#include "gtest/gtest.h"
+
+#ifdef GetObject
+#  undef GetObject
+#endif
+
+using namespace cppmicroservices;
+
+class BundleTrackerMethodTest : public ::testing::Test
+{
+protected:
+  Framework framework;
+  BundleContext context;
+  BundleTracker<>::BundleStateMaskType all_states =
+    BundleTracker<>::CreateStateMask(Bundle::State::STATE_ACTIVE,
+                                     Bundle::State::STATE_INSTALLED,
+                                     Bundle::State::STATE_RESOLVED,
+                                     Bundle::State::STATE_STARTING,
+                                     Bundle::State::STATE_STOPPING,
+                                     Bundle::State::STATE_UNINSTALLED);
+
+public:
+  BundleTrackerMethodTest()
+    : framework(FrameworkFactory().NewFramework()){};
+
+  ~BundleTrackerMethodTest() override = default;
+
+  void SetUp() override
+  {
+    framework.Start();
+    context = framework.GetBundleContext();
+  }
+
+  void TearDown() override
+  {
+    framework.Stop();
+    framework.WaitForStop(std::chrono::milliseconds::zero());
+  }
+};
+
+TEST_F(BundleTrackerMethodTest, CanCreateTracker)
+{
+  ASSERT_NO_THROW(BundleTracker bundleTracker(context, all_states))
+    << "Creation of BundleTracker failed";
+}
+
+TEST_F(BundleTrackerMethodTest, GetBundlesWorks)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  std::vector<Bundle> bundles = bundleTracker.GetBundles();
+  EXPECT_EQ(0, bundles.size())
+    << "GetBundles() should return an empty vector before Open()";
+
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundleMain = cppmicroservices::testing::GetBundle("main", context);
+  Bundle bundleSys =
+    cppmicroservices::testing::GetBundle("system_bundle", context);
+  Bundle bundleA = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  bundles = bundleTracker.GetBundles();
+
+  bool mainTracked =
+    std::find(bundles.begin(), bundles.end(), bundleMain) != bundles.end();
+  bool sysTracked =
+    std::find(bundles.begin(), bundles.end(), bundleSys) != bundles.end();
+  bool bundleATracked =
+    std::find(bundles.begin(), bundles.end(), bundleA) != bundles.end();
+
+  EXPECT_TRUE(mainTracked) << "GetBundles() should include the main bundle";
+  EXPECT_TRUE(sysTracked) << "GetBundles() should include the system bundle";
+  EXPECT_TRUE(bundleATracked) << "GetBundles() should include the test bundle";
+
+  bundleTracker.Close();
+  bundles = bundleTracker.GetBundles();
+  EXPECT_EQ(0, bundles.size())
+    << "GetBundles() should return an empty vector after Close()";
+}
+
+TEST_F(BundleTrackerMethodTest, GetObjectWorks)
```

Comment:
minor; rename testpoint name - remove "Works" to be more consistent with other test point names

Reply:
Done.